SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using SkietbaanBE.Models;
+
+namespace SkietbaanBE.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/Competition")]
+
+    public class CompetitionController : Controller
+    {
+        private ModelsContext _context;
+        public CompetitionController(ModelsContext db)
+        {
+            _context = db;
+        }
+ //----------------------------------------------------------------------------------------------------------------------------------------
+        /** The method to return an array of competition objects that hold the true Status*/
+        // GET: api/Competition
+        [HttpGet]
+        public IEnumerable<Competition> Get()
+        {
+            //get the competitions where(Status == true)
+            var competitionIDsQuery = from Comp in _context.Competitions
+                                      where Comp.Status == true
+                                      select Comp;
+            List<Competition> competitionsIDs = competitionIDsQuery.ToList<Competition>();
+            // return _context.Competitions.ToArray<Competition>();
+            return competitionsIDs;
+        }
+//----------------------------------------------------------------------------------------------------------------------------------------
+        //Getting the competition by ID
+        // GET: api/Competition/5
+        [HttpGet("{id}")]
+        public async Task<Competition> CompetitionGet(int id)
+        {
+            return await _context.Competitions.FindAsync(id);
+        }
+ //----------------------------------------------------------------------------------------------------------------------------------------
+        //Getting the competition by ID
+        // GET: api/Competition/5
+        [HttpGet("{Name}")]
+        public async Task<Competition> CompetitionGet(string Name)
+        {
+            return await _context.Competitions.FindAsync(Name);
+        }
+//----------------------------------------------------------------------------------------------------------------------------------------
+        //posting the competition to the competition table
+        // POST: api/Competition
+        [HttpPost]
+        public async Task<IActionResult> addCompetition([FromBody]Competition comp)
+        {
+            if (ModelState.IsValid)
+            {
+                Competition dbComp = null; //assume the competition does not exist
+
+                dbComp = _context.Competitions.FirstOrDefault(x => x.Name == comp.Name);
```

Comment:
Instead of 
`Competition dbComp = null;`
you can do this 
`dbComp = _context.Competitions.FirstOrDefault(x => x.Name == comp.Name);`
using var is much better for future updates

Reply:
and I suggest for the naming conventions, if you looking for someone that already exists in the database you can do something like this
`existingCompetition`
and in the parameter you can use 
`competition`